src/resolve-rc.js

```diff
@@ -27,6 +29,10 @@ const find = function find(start, rel) {
 
 module.exports = function(loc, rel) {
   rel = rel || ".babelrc";
+  const cacheKey = "" + loc + "/" + rel;
+  if (cacheKey in cache) {
+    return cache[cacheKey];
+  }
 
-  return find(loc, rel);
+  return (cache[cacheKey] = find(loc, rel));
```

Comment:
Done deal. I wasn't sure whether template strings were allowed by the local code style. :)